pokemon/damage.js

```diff
@@ -0,0 +1,154 @@
+class Pokemon {
+  constructor(pName,pType,pAttack,pDefense) {
+    this._pName = pName;
+    this._pType = pType;
+    this._pAttack = pAttack;
+    this._pDefense = pDefense;
+  }
+
+  get pName() {
+    return this._pName;
+  }
+  get pType() {
+    return this._pType;
+  }   
+
+  get pAttack(){
+    return this._pAttack;
+  }
+
+  get pDefense(){
+    return this._pDefense
+  }
+}
+
+
+function damage(pPokemon1,pPokemon2){
+  let resultdamage = 50 * (pPokemon1.pAttack/pPokemon2.pDefense) * effective(pPokemon1.pType,pPokemon2.pType);
+  return resultdamage;
+}
+
+
+function effective(pTypePokemon1,pTypePokemon2){
+  switch(pTypePokemon1){
```

Comment:
on js try to avoid "switch" statement, read these articles as reference:

- http://www.nikola-breznjak.com/blog/javascript/converting-a-javascript-switch-statement-into-a-function-lookup/
- https://toddmotto.com/deprecating-the-switch-statement-for-object-literals/